test/e2e/testnet/testnet.go

```diff
@@ -136,46 +129,72 @@ func (t *Testnet) CreateTxClients(ctx context.Context,
 	return nil
 }
 
-// CreateTxClient creates a txsim node and sets it up
-// name: name of the txsim knuu instance
-// version: version of the txsim docker image to be pulled from the registry
-// specified by txsimDockerSrcURL
-// seed: seed for the txsim
-// sequences: number of sequences to be run by the txsim
-// blobRange: range of blob sizes to be used by the txsim in bytes
-// pollTime: time in seconds between each sequence
-// resources: resources to be allocated to the txsim
-// grpcEndpoint: grpc endpoint of the node to which the txsim will connect and send transactions
+// CreateTxClient creates a txsim node and sets it up.
 func (t *Testnet) CreateTxClient(
 	ctx context.Context,
-	name, version string,
-	sequences int,
+	// name of the txsim knuu instance
+	name string,
+	// version of the txsim docker image to be pulled from the registry
+	version string,
+	// sequences: number of sequences to be run by the txsim
+	blobSequences int,
+	// blobRange: range of blob sizes to be used by the txsim in bytes
 	blobRange string,
 	blobPerSequence int,
+	// resources: resources to be allocated to the txsim
 	resources Resources,
+	// grpcEndpoint: grpc endpoint of the node to which the txsim will connect and send transactions
 	grpcEndpoint string,
+	// upgradeSchedule is a map from height to version, specifying the version to upgrade to at the given height
+	upgradeSchedule map[int64]uint64,
 ) error {
-	// create an account, and store it in a temp directory and add the account as genesis account to
-	// the testnet
 	txsimKeyringDir := filepath.Join(os.TempDir(), name)
-	log.Info().
-		Str("name", name).
-		Str("directory", txsimKeyringDir).
-		Msg("txsim keyring directory created")
-	_, err := t.CreateAccount(name, 1e16, txsimKeyringDir)
+	defer os.RemoveAll(txsimKeyringDir)
+
+	config := encoding.MakeConfig(app.ModuleEncodingRegisters...).Codec
+	txsimKeyring, err := keyring.New(app.Name, keyring.BackendTest, txsimKeyringDir, nil, config)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create keyring: %w", err)
 	}
 
-	// Create a txsim node using the key stored in the txsimKeyringDir
-	txsim, err := CreateTxClient(ctx, name, version, grpcEndpoint, t.seed,
-		sequences, blobRange, blobPerSequence, 1, resources, txsimRootDir, t.knuu)
+	key, _, err := txsimKeyring.NewMnemonic(name, keyring.English, "", "", hd.Secp256k1)
```

Comment:
[Question] Curious to know why CreateTxClient is not used here? any blocker?

Reply:
`CreateTxClient` gets used later in this function.